packages/babel-plugin-proposal-private-methods/src/index.js

```diff
@@ -3,20 +3,255 @@
 import { declare } from "@babel/helper-plugin-utils";
 import {
   createClassFeaturePlugin,
+  enableFeatureIfCompilingFields,
   FEATURES,
 } from "@babel/helper-create-class-features-plugin";
+import ReplaceSupers, {
+  environmentVisitor,
+} from "@babel/helper-replace-supers";
+import { isRequired } from "@babel/helper-compilation-targets";
 
 export default declare((api, options) => {
   api.assertVersion(7);
 
-  return createClassFeaturePlugin({
-    name: "proposal-private-methods",
+  if (!api.targets || isRequired("proposal-class-properties", api.targets())) {
```

Comment:
@developit was proposing to add something like `api.targets.support(name)` to the RFC, so that this line then becomes

```js
  if (!api.targets || api.targets.support("proposal-class-properties")) {
```